doctorapp/src/components/LoginForm.vue

```diff
@@ -0,0 +1,129 @@
+<template>
+  <div class="form-container">
+    <h1>DoctorApp</h1>
+    <form @submit.prevent="login">
+      <div class="input-group">
+        <label for="solid-pod-url">Solid Pod URL:</label>
+        <input type="url" id="solid-pod-url" v-model="solidPodUrl" required>
+        <p v-if="showWarning" class="warning">Invalid Solid Pod URL. Please check and try again.</p>
+      </div>
+      <div class="btn-group">
+        <button @click.prevent="setDefaultIdp" :disabled="isLoading">No Pod?</button>
+        <button type="submit" :disabled="isLoading">
+          <span v-if="isLoading">Loading...</span>
+          <span v-else>Login</span>
+        </button>
+      </div>
+    </form>
+  </div>
+</template>
+
+<script lang="ts">
+import { ref } from 'vue';
+import {store} from '@/store';
+
+export default {
+  setup(props, { emit }) {
```

Comment:
```suggestion
<script setup lang="ts">
import { ref } from 'vue';
import {store} from '@/store';

```

@BT-Creator and I are doing the Composition API, and this is Options API. Honestly, this is purely personal flavour, and thus you don't need to change it everywhere. But I would do it here because here there's only a `setup`.